### PR TITLE
Gravity generators now stop ship maneuver effects.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
As per title. If a gravity generator is on and producing gravity, you won't be dropped/thrown around anymore during maneuvers.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/overmap/ships/ship.dm [hunk 1]
```diff
@@ -276,10 +276,13 @@ var/const/OVERMAP_SPEED_CONSTANT = (1 SECOND)
 	forceMove(get_step(src, dir_to_move))
 	for(var/mob/living/L in living_mob_list)
 		if(L.z in map_z)
-			to_chat(L, SPAN_DANGER("<font size=4>The ship rapidly inclines under your feet!</font>"))
-			if(!L.buckled_to)
-				var/turf/T = get_step_away(get_turf(L), get_step(L, new_dir), 10)
-				L.throw_at(T, 10, 10)
+			if(!gravity_generator?.gravity_is_on)
+				to_chat(L, SPAN_DANGER("<font size=4>The ship rapidly inclines under your feet!</font>"))
+				if(!L.buckled_to)
```
Comment (SleepyGemmy) [inline code/modules/overmap/ships/ship.dm:281]: check if anchored?

File: code/modules/overmap/ships/ship.dm [hunk 2]
```diff
@@ -308,16 +311,19 @@ var/const/OVERMAP_SPEED_CONSTANT = (1 SECOND)
 	update_icon()
 	for(var/mob/living/L in living_mob_list)
 		if(L.z in map_z)
-			to_chat(L, SPAN_DANGER("The ship rapidly turns under your feet!"))
-			if(!L.buckled_to)
-				L.Weaken(3)
+			if(!gravity_generator?.gravity_is_on)
+				to_chat(L, SPAN_DANGER("The ship rapidly turns beneath you!"))
+				if(!L.buckled_to)
```
Comment (SleepyGemmy) [inline code/modules/overmap/ships/ship.dm:316]: check if anchored?

**Review by SleepyGemmy — APPROVED** [review]

File: code/modules/overmap/ships/ship.dm (hunk 1 above)
Comment (NonQueueingMatt) [reply code/modules/overmap/ships/ship.dm:281]: already done above

File: code/modules/overmap/ships/ship.dm (hunk 2 above)
Comment (NonQueueingMatt) [reply code/modules/overmap/ships/ship.dm:316]: already done above

**Pull request merged**